### ISSUE TITLE
Zeep dont handle restriction on simpleTypes?

### ISSUE THREAD

**Issue by 00cc00** [mvantellingen/python-zeep#513]
1. The version of zeep: 2.2.0 (python 2.7)
2. The WSDL you are using
```
<xs:simpleType name="RollenEnum">
    <xs:restriction base="xs:string">
        <xs:enumeration value="VersichertePerson"/>
        <xs:enumeration value="Versicherungsnehmer"/>
        <xs:enumeration value="Beitragszahler"/>
        <xs:enumeration value="BezugsberechtigterImTodesfall"/>
        <xs:enumeration value="Ehepartner"/>
        <xs:enumeration value="GesetzlicherVertreter"/>
    </xs:restriction>
</xs:simpleType>
```
Hello there,

I'm only using the zeep library to parse WSDL files, so I don't use it as a real SOAP Client.

I need this to map it against own datamodels. However, I can't figure out if zeep handles restrictions in simpleTypes. The Type 'RollenEnum' is after parsing a 'String' BuiltinType. This is good in the first place. But I also need to map the restrictions like Enums and so on. Doesn't zeep support this or do I miss something?

I also tried to create Objects of this Type where I was able to input other Values than defined in the Enumeration. 

Another example would be this:
```
<xs:simpleType name="someType">
    <xs:restriction base="xs:integer">
        <minInclusive value="0"/>
        <maxInclusive value="100"/>
    </xs:restriction>
</xs:simpleType>
```
Again, if I create a Type of 'someType' like so: ``someType(120)`` I don't get any Exception or something although the Value isn't valid like the Schema said. 

So again my Question: Do I miss something or is this just not supported?

Thank you very much in advance for the help!

**Comment by Hansa064** [comment]
I've got the same problem. 
I need to be able to use restrictions to consume, validate and create WebServices.

The problem here seems to be, that the [SchemaVisitor](https://github.com/mvantellingen/python-zeep/blob/master/src/zeep/xsd/visitor.py#L632) is reading the base of the restriction and returns and does not consume the children of the restriction tag. Here, handlers for the different restriction types need to be implemented.

**Comment by mvantellingen** [comment]
Yeah the data validation is still not really implemented properly. I've started adding some validation based on if something is required or not but thats it. It's something I want to focus on next (and can definitely use help on this)

**Comment by mvantellingen** [comment]
There are no xsd validation libraries for python. Originally I intended to rely on link for validation but that doesn’t support dad linking at the wsdl level. 

Validation shouldn’t be too hard, just haven’t given it priority

**Comment by sanjayhallan** [comment]
+1 for this issue
I am moving from suds to zeep as google has moved their googleads-python-lib to zeep from suds. 
I'm trying to get a enum type which works but has no attributes :-), so not much use
I will need to use the suds client to handle enum types - not ideal having two soap clients hanging around. 
Cheers
Sanjay

**Comment by carlosmaguda** [comment]
@mvantellingen I have urgency on this matter. Kindly contact me so that we can work something out.

**Comment by set92** [comment]
I suppose there are no development in this feature no? So the temporal fix I suppose is to use other library? Someone has a recommendation? Even if it is in another language, because I think suds is worse than Zeep, and there isn't a better one than Zeep in Python

**Comment by ManuelAngel99** [comment]
Hi! Is there any progress on this issue?
